module/Company/src/Company/Controller/AdminController.php

```diff
@@ -0,0 +1,311 @@
+            }
+        }
+        $companyForm->setAttribute('action',
+                                   $this->url()->fromRoute('admin_company/default',
+                                                           array('action' => 'addCompany'))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $companyForm,
+        ));
+
+        return $vm;
+    }
+    public function deletePacketAction()
+    {
+        $companyService = $this->getCompanyService();
+        $packetID = $this->params('packetID');
+        $companyName = $this->params('slugCompanyName');
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            $del = $request->getPost('del', 'No');
+            if ($del == 'Yes') {
+                $companyService->deletePacket($packetID);
+            }
+
+            return $this->redirect()->toRoute('admin_company/editCompany', array('slugCompanyName' => $companyName));
+        }
+
+        return new ViewModel(array(
+            'packet' => $companyService->getEditablePacket($packetID),
+            'slugName' => $companyName,
+            'translator' => $companyService->getTranslator(),
+        ));
+    }
+    public function deleteCompanyAction()
+    {
+        $companyService = $this->getCompanyService();
+        $slugName = $this->params('slugCompanyName');
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            $del = $request->getPost('del', 'No');
+            if ($del == 'Yes') {
+                $companyService->deleteCompaniesWithSlug($slugName);
+            }
+
+            return $this->redirect()->toRoute('admin_company');
+        }
+
+        return new ViewModel(array(
+            'companies' => $companyService->getEditableCompaniesWithSlugName($slugName),
+            'translator' => $companyService->getTranslator(),
+        ));
+    }
+
+    public function addJobAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyName = $this->params('slugCompanyName');
+        $packetId = $this->params('packetID');
+
+        $companyForm = $companyService->getJobForm();
+
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($jobName)) {
+                $jobName = $request->getPost()['slugName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm = $companyService->getJobForm();
+            $companyForm->setData($request->getPost());
+
+            if ($companyForm->isValid()) {
+                $job = $companyService->insertJobIntoPacketID($packetId);
+                $job->exchangeArray($request->getPost());
+                $companyService->saveCompany();
+
+                return $this->redirect()->toRoute('admin_company/editCompany/editPacket/editJob',
```

Comment:
Please format to the PSR standard:

http://www.php-fig.org/psr/psr-2/#4-6-method-and-function-calls